Packages/Core/Runtime/ValueLists/AtomValueList.cs

```diff
@@ -42,6 +44,59 @@ public abstract class AtomValueList<T, E> : BaseAtomValueList, IList<T>
         [SerializeField]
         protected List<T> list = new List<T>();
 
+        private List<T> _initial;
+        
+        
+#if UNITY_EDITOR
+        /// <summary>
+        /// Set of all AtomVariable instances in editor.
+        /// </summary>
+        private static HashSet<AtomValueList<T, E>> _instances = new HashSet<AtomValueList<T, E>>();
+#endif
+        
+        protected virtual void OnEnable()
+        {
+#if UNITY_EDITOR
+            if (EditorSettings.enterPlayModeOptionsEnabled)
+            {
+                _instances.Add(this);
+
+                EditorApplication.playModeStateChanged -= HandlePlayModeStateChange;
+                EditorApplication.playModeStateChanged += HandlePlayModeStateChange;
+            }
+#endif
+        }
+        
+#if UNITY_EDITOR
+        private static void HandlePlayModeStateChange(PlayModeStateChange state)
+        {
+            if (state == PlayModeStateChange.ExitingEditMode) // BEFORE any GO is initialized:
+            {
+                foreach (var instance in _instances)
+                {
+                    if (instance == null)
+                    {
+                        _instances.Remove(instance);
+                        continue;
+                    }
```

Comment:
This could cause an `InvalidOperationException` when removing an item in the set while iterating over it. I get the need though to remove an instance that is no longer there. I believe we need a strategy to do this in this file, but also for `AtomVariable.cs` and `AtomEvent.cs`. Maybe we can look into [this](https://forum.unity.com/threads/ondestroy-and-ondisable-are-not-called-when-deleting-a-scriptableobject-file.1129220/#post-7259671) and see if we can approach it in a similar fashion. 

Furtermore, just checking for null won't suffice. You can still get this error:
`MissingReferenceException: The object of type 'IntVariable' has been destroyed but you are still trying to access it.` Got the above while testing when I: 
1) Enabled enter play mode in `unity-atoms-examples`. Eg: 
<img width="365" alt="Skärmavbild 2023-08-03 kl  07 00 09" src="https://github.com/unity-atoms/unity-atoms/assets/10260896/1d8d1276-2c77-42d0-9c24-cfc97d7878bb">
2) Added debug statements  in `HandlePlayModeStateChange` in `AtomVariable.cs` like this: 
```
#if UNITY_EDITOR
        private static void HandlePlayModeStateChange(PlayModeStateChange state)
        {
            Debug.Log($"HandlePlayModeStateChange: {state}. Instances: {_instances.Count}");
            if (state == PlayModeStateChange.ExitingEditMode)  // BEFORE any GO is initialized:
            {
                foreach (var instance in _instances)
                {
                    instance.SetInitialValues();
                    Debug.Log($"HandlePlayModeStateChange instance: {instance.name}");
                }
            }
            else if (state == PlayModeStateChange.EnteredPlayMode)  // within/end of the first frame
            {
                foreach (var instance in _instances)
                {
                    instance.TriggerInitialEvents();
                    Debug.Log($"HandlePlayModeStateChange instance: {instance.name}");
                };
            }
        }
#endif
```